doc/guides/modules.rst

```diff
@@ -253,9 +263,43 @@ Where 23 in the example above indicated the pull request number submitted to the
 *my_module* repository. Once the module changes are reviewed and merged, the
 revision needs to be changed to the commit hash from the module repository.
 
-
+Upmerging forks
+===============
+
+When a module is in fact a fork of an existing, external project that is hosted
+on GitHub or elsewhere, it is often necessary to upmerge it: update the copy
+that is hosted under the `zephyrproject-rtos`_ GitHub organization with the
+latest changes from the external upstream repository.
+
+The procedure for upmerging module repositories that are forks of external
+projects is constrained by the following two factors:
+
+* History must **never** be rewritten in any repository that is referenced in
+  the `west.yml`_ manifest file, since that could break older revisions of the
+  repository set as a whole. This therefore excludes rebasing the module
+  repository and keeping the zephyr-specific changes always on top.
+* GitHub does not deal correctly with merge commits contained inside Pull
+  Requests, which prevents us from using regular ``git merge`` operations
+  without having to resort to pushing directly to the repository.
+
+Due to the above, this is the procedure that you must use in order to upmerge a
+module repository with the latest changes from the external upstream
+repository:
+
+#. Make a note of the current upstream revision that the module repository is
+   currently upmerged to.
+#. Take the changes from the current to the latest upstream revision and apply
+   them to the current ``master`` branch of the module repository. There are
+   several ways of achieving this  It is important to note that the
+   module repository may not share a common Git history with the original
+   upstream repository, so using ``git diff`` and ``git apply`` might be the
+   simplest course of action.
+#. Submit a pull request against the module and zephyr (manifest) repository as
+   described in :ref:`changes_to_existing_module`.
 
 .. _CMake list: https://cmake.org/cmake/help/latest/manual/cmake-language.7.html#lists
 .. _add_subdirectory(): https://cmake.org/cmake/help/latest/command/add_subdirectory.html
 
 .. _GitHub issues: https://github.com/zephyrproject-rtos/zephyr/issues
+.. _zephyrproject-rtos: https://github.com/zephyrproject-rtos
+.. _west.yml: https://github.com/zephyrproject-rtos/zephyr/blob/master/west.yml
```

Comment:
Why are you removing the `:zephyr_file:` uses?

That custom role has a hope of being extended to not go stale by emitting links for release documentation that don't point at `master`. This doesn't.

See the discussion here for details on my concern: https://github.com/zephyrproject-rtos/zephyr/pull/19123#discussion_r324037850